classes/class-dintero-checkout-subscription.php

```diff
@@ -316,7 +305,6 @@ public function handle_redirect_from_change_payment_method( $subscription_id ) {
 			}
 
 			$subscription->add_order_note( $message );
-			$subscription->save();
```

Comment:
Doesn't  [`add_order_note`](https://woocommerce.github.io/code-reference/files/woocommerce-includes-class-wc-order.html#source-view.2020) require a `save` call? If yes, there are other places where the `save` call has been omitted in this PR.